includes/wc-core-functions.php

```diff
@@ -793,13 +793,25 @@ function wc_get_page_children( $page_id ) {
 /**
  * Flushes rewrite rules when the shop page (or it's children) gets saved.
  */
-function flush_rewrite_rules_on_shop_page_save( $post_id ) {
+function flush_rewrite_rules_on_shop_page_save() {
+	// Check if is in edit page.
+	if ( 'page' !== get_current_screen()->id ) {
```

Comment:
Check `get_current_screen()` is an object first; this has caused errors in the past.